### ISSUE TITLE
--log-level panic while writing log_level to config (wrong type - &str / LogLevel)

### ISSUE THREAD

**Issue by Nindouja** [rust-starter/rust-starter#50]
### Rust Version

rustc 1.79.0 (129f3b996 2024-06-10)

### Affected Version of clap

```

[[package]]
name = "clap"
version = "4.5.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "5db83dced34638ad474f39f250d7fea9598bdd239eaced1bdf45d597da0f433f"
dependencies = [
 "clap_builder",
 "clap_derive",
]

[[package]]
name = "clap_builder"
version = "4.5.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "f7e204572485eb3fbf28f871612191521df159bc3e15a9f5064c66dba3a8c05f"
dependencies = [
 "anstream",
 "anstyle",
 "clap_lex",
 "strsim",
]

[[package]]
name = "clap_complete"
version = "4.5.6"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "fbca90c87c2a04da41e95d1856e8bcd22f159bdbfa147314d2ce5218057b0e58"
dependencies = [
 "clap",
]

[[package]]
name = "clap_derive"
version = "4.5.5"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "c780290ccf4fb26629baa7a1081e68ced113f1d3ec302fa5948f1c381ebf06c6"
dependencies = [
 "heck",
 "proc-macro2",
 "quote",
 "syn",
]

[[package]]
name = "clap_lex"
version = "0.7.1"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "4b82cf0babdbd58558212896d1a4272303a57bdb245c2bf1147185fb45640e70"

[[package]]
name = "cli"
version = "2.0.0"
dependencies = [
 "assert_cmd",
 "clap",
 "clap_complete",
 "core",
 "predicates",
 "utils",
]
```

### Bug or Feature Request Summary

It seems the log level override from cli args is not working. It's trying to assign a &str into a LogLevel type in the config struct

### Steps to Reproduce the issue

Run a command with `--log-level eror` from a freshly cloned project. Examples below

### Sample Code or Link to Sample Code

`cargo run -- --log-level error config` or  `cargo build && ./target/debug/rust-starter --log-level error config`

### Debug output

```log
/home/toto/Documents/Code/rust-starter/target/debug/rust-starter --log-level error config
Backtrace (most recent call last):
  File "/home/nindouja/Documents/Code/rust-starter/src/main.rs:35", in rust_starter::main
    cli::cli_match()?;
  File "/home/nindouja/Documents/Code/rust-starter/cli/src/lib.rs:91", in cli::cli_match
    AppConfig::merge_args(matches)?;
  File "/home/nindouja/Documents/Code/rust-starter/utils/src/app_config.rs:65", in utils::app_config::AppConfig::merge_args
    let value: &str = *args.get_one("log_level").unwrap_or(&"info");
  File "/home/nindouja/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.7/src/parser/matches/arg_matches.rs:115", in clap_builder::parser::matches::arg_matches::ArgMatches::get_one
    MatchesError::unwrap(id, self.try_get_one(id))
  File "/home/nindouja/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.7/src/parser/error.rs:32", in clap_builder::parser::error::MatchesError::unwrap
    panic!("Mismatch between definition and access of `{id}`. {err}",)

The application panicked (crashed).
  Mismatch between definition and access of `log_level`. Could not downcast to &str, need to downcast to utils::types::LogLevel
in utils/src/app_config.rs:65
thread: main

```

**Comment by Nindouja** [comment]
If someone can guide me to the resolution, I'd be happy to propose a PR

**Comment by omarabid** [comment]
Fixed in #ec1f6db. Re-open if you still have this error.
